wgengine/userspace.go

```diff
@@ -1174,6 +1186,8 @@ func (e *userspaceEngine) linkChange(delta *netmon.ChangeDelta) {
 			if dnsCfg != nil {
 				if err := e.dns.Set(*dnsCfg); err != nil {
 					e.logf("wgengine: error setting DNS config after major link change: %v", err)
+				} else if err := e.reconfigureVPNIfNecessary(); err != nil {
```

Comment:
up above it says "hacky workaround for Linux DNS" and then includes ios and android and darwin 😢 

Maybe time to update that comment to justify why Android at least needs this? And leave TODOs for justifying ios and darwin?